php-checks/src/main/java/org/sonar/php/checks/OverridingMethodSimplyCallParentCheck.java

```diff
@@ -90,25 +99,58 @@ private void checkMethod(MethodDeclarationTree method, String superClass) {
     }
   }
 
-  private void checkExpression(@Nullable ExpressionTree expressionTree, MethodDeclarationTree method, String superClass) {
+  private void checkExpression(@Nullable ExpressionTree expressionTree, MethodDeclarationTree method, ClassSymbol superClass) {
     if (expressionTree != null && expressionTree.is(Kind.FUNCTION_CALL)) {
       FunctionCallTree functionCallTree = (FunctionCallTree) expressionTree;
 
       if (functionCallTree.callee().is(Kind.CLASS_MEMBER_ACCESS)) {
         MemberAccessTree memberAccessTree = (MemberAccessTree) functionCallTree.callee();
-
         String methodName = method.name().text();
-        boolean sameMethodName = memberAccessTree.member().toString().equals(methodName);
 
-        if (isSuperClass(memberAccessTree.object(), superClass) && sameMethodName && sameArguments(functionCallTree, method)) {
-          String message = String.format(MESSAGE, methodName);
-          context().newIssue(this, method.name(), message);
+        boolean isCallingSuperclassMethodWithSameNameAndArguments = isSuperClassReference(memberAccessTree.object(), superClass.qualifiedName().toString()) &&
+          memberAccessTree.member().toString().equals(methodName) &&
+          isFunctionCalledWithSameArgumentsAsDeclared(functionCallTree, method);
+        if (isCallingSuperclassMethodWithSameNameAndArguments) {
+          boolean duplicatesDeclarationFromSuper = Stream.iterate(superClass, Objects::nonNull, c -> c.superClass().orElse(null))
```

Comment:
Nice way for iterate over class hierarchy, however stream in the stream is always harder to read